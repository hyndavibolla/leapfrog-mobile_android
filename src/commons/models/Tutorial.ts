import { ReactNode } from 'react';
import { StyleProp, ViewStyle } from 'react-native';

export interface Tutorial {
  isParent?: boolean;
  isHidden?: boolean;
  parentId?: number;
  scrollToDown?: number;
}

export interface TutorialElementPosition extends Tutorial {
  x: number;
  y: number;
  children: ReactNode;
  style: StyleProp<ViewStyle>;
}
