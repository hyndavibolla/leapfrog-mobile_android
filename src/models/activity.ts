import { IOffer } from './offer';

import { DateLike } from './general';

export enum Status {
  AVAILABLE = 'available',
  PENDING_CONFIRMATION = 'pendingConfirmation',
  POINT_BURN = 'pointBurn'
}

export enum Type {
  TRANSACTION = 'transaction',
  ADJUSTMENT = 'adjustment',
  REDEMPTION = 'redemption',
  REDEEM = 'redeem',
  EXPIRY = 'pointsexpiry',
  AVAILABLE = 'pointsavailable',
  RETURN = 'return',
  OTHER = 'other'
}

export interface IActivity {
  points: number;
  activityType: Type;
  requestorName: string;
  offers: IOffer[];
  timestamp: DateLike;
  grossSpend?: number;
  brandDetails?: {
    brandName: string;
    brandShortDescription: string;
    brandLogo: string;
  };
  txnId: string;
  activityDetails?: {
    partnerId: string;
    giftCardName: string;
    giftCardValue?: number;
  };
}

export const getFallbackActivity = (): IActivity => ({
  points: 0,
  activityType: Type.TRANSACTION,
  requestorName: '...',
  offers: [],
  timestamp: undefined,
  grossSpend: undefined,
  txnId: ''
});

export const getStatusDisplayText = (activityType: string, offer: IOffer): string => {
  if (!offer.pointStartDate || new Date(offer.pointStartDate).getTime() <= Date.now()) return 'Available';
  if (new Date(offer.pointStartDate).getTime() > Date.now()) return 'Pending';
  return activityType;
};
