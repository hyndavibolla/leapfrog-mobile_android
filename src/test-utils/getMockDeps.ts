import { ENV } from '_constants';
import { GeneralModel, MissionModel } from '_models';
import { RUNTIME_ENV } from '_models/general';
import { ICrashlyticsService } from '_services/CrashlyticsService';
import { IHttpService } from '_services/HttpService';
import { IPerformanceTrackingService } from '_services/PerformanceTrackingService';
import { getInitialState } from '_state_mgmt/GlobalState';
import {
  getUser,
  getGame,
  getActivity_1,
  getActivity_2,
  getActivity_3,
  getMissionCategory_1,
  getMissionCategory_2,
  getMissionCategory_3,
  getMissionKeyword_1,
  getMissionKeyword_2,
  getMissionKeyword_3,
  getMission_1,
  getRewardConfig_1,
  getSlideObject_1,
  getSlideObject_2,
  getSlideCategoryIdList,
  getStreak_1,
  getQuestion_1,
  getQuestion_2,
  getQuestion_3,
  getAnswer_1,
  getAnswer_2,
  getAnswer_3,
  getNewOnMaxMessage_1,
  getNewOnMaxMessage_2,
  getNewOnMaxMessage_3,
  getSailthruMessage_1,
  getLinkedCards_1,
  getLinkedCards_2,
  getLinkedCards_3,
  getLinkedCards_4,
  getNewOnMaxMessage_4,
  getGiftCard,
  getGiftCardDetail,
  getGiftCardBalance
} from '_test_utils/entities';
import { createUUID } from '_utils/create-uuid';

export const getMockDeps = (initialStateGetter = getInitialState): GeneralModel.Deps => ({
  apiService: {
    isTokenInvalid: false,
    getDeviceIdAsync: async () => createUUID(),
    isTokenSetCurrentlyReady: () => true,
    setTokenSet: jest.fn(),
    authenticate: jest
      .fn()
      .mockResolvedValue({ accessToken: 'accessToken', refreshToken: 'refreshToken', accessTokenExpiryTime: 'accessTokenExpiryTime', deviceId: 'deviceId' }),
    enrollCard: jest.fn().mockResolvedValue(undefined),
    fetchRefId: jest.fn().mockResolvedValue('test-ref-id'),
    fetchUserProfile: jest.fn().mockResolvedValue(getUser()),
    fetchGameState: jest.fn().mockResolvedValue(getGame()),
    fetchRewardConfig: jest.fn().mockResolvedValue(getRewardConfig_1()),
    fetchActivityHistory: jest.fn().mockResolvedValue([getActivity_1(), getActivity_2(), getActivity_3()]),
    fetchGiftCardList: jest.fn().mockResolvedValue([getGiftCard()]),
    fetchGiftCardDetail: jest.fn().mockResolvedValue(getGiftCardDetail()),
    fetchGiftCardBalance: jest.fn().mockResolvedValue(getGiftCardBalance()),
    fetchWalletStats: jest.fn().mockResolvedValue({
      totalLifetimeEarnedPoints: 15550,
      monthlySummaries: {
        '2020-02': 1500,
        '2020-03': 2000,
        '2020-04': 100,
        '2020-05': 950,
        '2020-06': 1000,
        '2020-07': 1500
      }
    }),
    fetchMissionCategoryList: jest.fn().mockResolvedValue([getMissionCategory_1(), getMissionCategory_2(), getMissionCategory_3()]),
    fetchMissionKeywordList: jest.fn().mockResolvedValue({
      [MissionModel.KeywordType.CATEGORY]: [getMissionKeyword_1().value],
      [MissionModel.KeywordType.BRAND]: [getMissionKeyword_2().value, getMissionKeyword_3().value]
    }),
    fetchMissionList: jest.fn(async ({ listType }) => ({ userId: 'buttonId', missions: [getMission_1()], title: 'mission endpoint title', listType })),
    registerDeviceInfo: jest.fn(),
    registerUserActivity: jest.fn(),
    logout: jest.fn(),
    getSYWOnlineToken: jest.fn().mockResolvedValue('test-token'),
    sendValidationEmail: jest.fn().mockResolvedValue(undefined),
    fetchStreakList: jest.fn().mockResolvedValue([getStreak_1()]),
    fetchPromptList: jest.fn().mockResolvedValue({
      promptQuestions: [getQuestion_1(), getQuestion_2(), getQuestion_3()],
      answeredQuestions: [getAnswer_1(), getAnswer_2(), getAnswer_3()]
    }),
    updatePromptList: jest.fn(),
    unregisterDeviceInfo: jest.fn(),
    fetchLinkedCardsList: jest.fn().mockResolvedValue({ linkedCards: [getLinkedCards_1(), getLinkedCards_2(), getLinkedCards_3(), getLinkedCards_4()] }),
    activateLocalOffer: jest.fn(),
    fetchAppleWalletPass: jest.fn().mockResolvedValue('test-pass'),
    updateCardStatus: jest.fn()
  } as any,
  appService: {
    getDeviceIdAsync: async () => createUUID()
  } as any,
  crashlyticsService: {
    initialize: jest.fn(),
    log: jest.fn(),
    recordError: jest.fn(),
    setCrashlyticsCollectionEnabled: jest.fn()
  } as ICrashlyticsService,
  httpService: {
    fetch: jest.fn()
  } as IHttpService,
  performanceTrackingService: {
    initialize: jest.fn(),
    measuredFetch: jest.fn(),
    setPerformanceCollectionEnabled: jest.fn()
  } as IPerformanceTrackingService,
  raiseService: {
    fetchSlideBrand: jest.fn().mockResolvedValue([getSlideObject_1(), getSlideObject_2()]),
    fetchSlideCategoryIdList: jest.fn().mockResolvedValue(getSlideCategoryIdList())
  } as any,
  nativeHelperService: {
    alert: {
      alert: jest.fn(),
      prompt: jest.fn()
    },
    appState: { addEventListener: jest.fn(), currentState: 'active' },
    storage: {
      get: jest.fn().mockResolvedValue(null),
      set: jest.fn().mockResolvedValue(null),
      remove: jest.fn().mockResolvedValue(null)
    },
    clipboard: {
      getString: jest.fn(),
      setString: jest.fn()
    },
    dimensions: {
      getWindowWidth: jest.fn().mockReturnValue(750),
      getWindowHeight: jest.fn().mockReturnValue(1000),
      isSmallDevice: false
    },
    deviceInfo: {
      getDeviceId: createUUID,
      getUniqueId: createUUID,
      getVersion: () => '0.0',
      getBuildNumber: () => '0.1',
      getDeviceModel: () => 'iPhone 11',
      getLocales: { languageTag: 'en-US' },
      getBaseOs: async () => 'iOS',
      getSystemVersion: () => '11.0',
      getDeviceToken: () => createUUID(),
      getDeviceTokenAsync: async () => createUUID(),
      getDeviceOsType: async () => 'iOS',
      getAppVersion: () => '0.0.0.1',
      getDeviceLanguage: () => 'en-US',
      getDeviceOsVersion: () => '11.0',
      getFCMTokenAsync: async () => createUUID(),
      getTimeZone: () => 'testing-timezone',
      getReadableVersion: () => '1.1.1.1',
      onFCMTokenRefresh: jest.fn()
    },
    netInfo: {
      getState: jest.fn().mockResolvedValue({ isConnected: true }),
      subscribeToConnectionChanges: jest.fn(),
      waitForConnection: jest.fn().mockResolvedValue(undefined)
    },
    linking: {
      openURL: jest.fn(() => Promise.resolve()),
      canOpenURL: jest.fn(),
      checkURLScheme: jest.fn().mockResolvedValue(true),
      getInitialURL: jest.fn(),
      addEventListener: jest.fn()
    },
    buttonSdk: {
      configure: jest.fn(),
      setUserId: jest.fn(),
      clearData: jest.fn(),
      purchaseRequest: jest.fn()
    },
    platform: {
      os: 'default',
      select: jest.fn(config => config?.default)
    },
    fireBaseMessaging: {
      getToken: jest.fn().mockReturnValue(createUUID())
    },
    hash: {
      buildSHA256: jest.fn().mockResolvedValue('hash')
    },
    restart: jest.fn(),
    cookieManager: {
      clearAll: jest.fn()
    },
    base64: {
      encode: jest.fn((...args) => `b64 encoded: ${JSON.stringify(args)}`),
      decode: jest.fn((...args) => `b64 decoded: ${JSON.stringify(args)}`)
    },
    getRuntimeEnv: jest.fn().mockResolvedValue(RUNTIME_ENV.DEV_TEST),
    messaging: {
      hasPermission: jest.fn().mockReturnValue(true),
      requestPermission: jest.fn()
    },
    pushNotificationsAuthStatus: {
      NOT_DETERMINED: -1,
      DENIED: 0,
      AUTHORIZED: 1,
      PROVISIONAL: 2
    },
    sailthru: {
      getMessages: jest
        .fn()
        .mockResolvedValue([getNewOnMaxMessage_1(), getNewOnMaxMessage_4(), getNewOnMaxMessage_2(), getNewOnMaxMessage_3(), getSailthruMessage_1()]),
      markMessageAsRead: jest.fn().mockResolvedValue(true),
      getInitialNotification: jest.fn(),
      onNotificationTapped: jest.fn(cb => {
        cb({});
        return () => {};
      }) as any
    },
    reactNativePermission: require('react-native-permissions/mock'),
    geolocation: {
      addListener: jest.fn(),
      getCurrentPosition: jest.fn(),
      removeListeners: jest.fn(),
      requestAuthorization: jest.fn(),
      setConfiguration: jest.fn(),
      startObserving: jest.fn(),
      stopObserving: jest.fn()
    }
  } as any,
  stateSnapshot: {
    set: jest.fn(),
    get: jest.fn().mockReturnValue(initialStateGetter())
  },
  logger: {
    log: jest.fn(),
    info: jest.fn(),
    debug: jest.fn(),
    warn: jest.fn(),
    error: jest.fn(),
    success: jest.fn(),
    color: jest.fn(),
    assert: jest.fn(),
    addEventListener: jest.fn(),
    removeEventListener: jest.fn(),
    format: jest.fn()
  } as any,
  eventTrackerService: {
    trackDataUser: jest.fn(),
    isTrackingEnabled: jest.fn().mockResolvedValue(true),
    rZero: {
      trackView: jest.fn(),
      trackEvent: jest.fn(),
      flush: jest.fn(),
      setUser: jest.fn()
    },
    forterSDK: {
      setDevLogsEnabled: jest.fn(),
      getDeviceUniqueID: jest.fn(cb => cb()),
      init: jest.fn((a, b, c) => c()),
      trackNavigation: jest.fn(),
      trackNavigationWithExtraData: jest.fn(),
      trackAction: jest.fn(),
      trackActionWithJSON: jest.fn(),
      fraudTrackNavigationEvent: jest.fn(),
      fraudTrackAction: jest.fn()
    },
    tealiumSDK: {
      track: jest.fn(),
      trackView: jest.fn()
    },
    appsFlyerSDK: {
      updateServerUninstallToken: jest.fn(),
      setCustomerUserId: jest.fn().mockImplementation((_, cb) => cb('success')),
      addDeepLinkEventListener: jest.fn(),
      removeDeepLinkEventListener: jest.fn(),
      clearDeepLinkEventListener: jest.fn(),
      trackConversionEvent: jest.fn()
    }
  } as any,
  ccpaService: {
    showConsentBanner: jest.fn().mockResolvedValue(null),
    showPreferenceCenter: jest.fn()
  } as any,
  remoteConfigService: {
    getValue: jest.fn(async key => ENV.REMOTE_CONFIG.DEFAULT_CONFIG[key]),
    fetchAndActivate: jest.fn().mockResolvedValue(null)
  } as any,
  authService: {
    auth: {
      refreshToken: jest.fn().mockResolvedValue(true)
    },
    webAuth: {
      authorize: jest.fn(),
      clearSession: jest.fn()
    }
  } as any
});
