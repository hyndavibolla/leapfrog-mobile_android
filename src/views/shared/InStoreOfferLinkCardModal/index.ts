export { default as InStoreOfferLinkCardModal, InStoreOfferLinkCardModalKey } from './InStoreOfferLinkCardModal';
