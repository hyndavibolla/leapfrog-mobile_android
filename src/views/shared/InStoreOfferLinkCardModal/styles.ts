import { css } from '@emotion/native';

import { COLOR, FONT_FAMILY, FONT_SIZE } from '../../../constants';

export const styles = {
  container: css`
    align-items: center;
  `,
  header: css`
    flex-direction: row;
    justify-content: center;
    align-items: center;
    margin-bottom: 24px;
  `,
  brandIcon: css`
    min-width: 60px;
    min-height: 60px;
  `,
  title: css`
    text-align: center;
    color: ${COLOR.BLACK};
  `,
  body: css`
    margin-top: 8px;
  `,
  bodyText: css`
    text-align: center;
    font-size: ${FONT_SIZE.SMALLER};
  `,
  plus: css`
    color: ${COLOR.BLACK};
    font-size: ${FONT_SIZE.MEDIUM};
    line-height: ${FONT_SIZE.HUGE};
    font-family: ${FONT_FAMILY.HEAVY};
    margin: 0 12px;
  `,
  linkBtnInnerContainer: css`
    padding: 16px 100px;
    margin-top: 40px;
  `,
  linkBtnText: css`
    font-family: ${FONT_FAMILY.MEDIUM};
    font-size: ${FONT_SIZE.SMALLER};
  `,
  dismissBtnInnerContainer: css`
    margin-top: 16px;
    background-color: transparent;
  `,
  dismissBtnText: css`
    color: ${COLOR.PRIMARY_BLUE};
    font-family: ${FONT_FAMILY.BOLD};
    font-size: ${FONT_SIZE.SMALLER};
  `
};
