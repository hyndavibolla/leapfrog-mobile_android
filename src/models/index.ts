import * as GeneralModel from './general';
import * as UserModel from './user';
import * as ActivityModel from './activity';
import * as OfferModel from './offer';
import * as GameModel from './game';
import * as MissionModel from './mission';
import * as RewardModel from './reward';
import * as StreakModel from './streak';
import * as PromptModel from './prompt';
import * as CardLink from './cardLink';

export { GeneralModel, UserModel, ActivityModel, OfferModel, GameModel, MissionModel, RewardModel, StreakModel, PromptModel, CardLink };
