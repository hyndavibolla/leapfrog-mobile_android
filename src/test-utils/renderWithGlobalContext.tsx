import React, { ReactElement } from 'react';
import { render } from '@testing-library/react-native';

import { getInitialState, GlobalProvider } from '../state-mgmt/GlobalState';
import { Deps, IGlobalState, IReducer } from '../models/general';
import { getMockDeps } from './getMockDeps';
import { getMockReducer } from './getMockReducer';

export const renderWithGlobalContext = (
  children: ReactElement,
  deps: Deps = getMockDeps(),
  initState: IGlobalState = getInitialState(),
  combinedReducers: IReducer<IGlobalState, any> = getMockReducer()
) => {
  const renderAPI = render(
    <GlobalProvider deps={deps} initState={initState} combinedReducers={combinedReducers}>
      {children}
    </GlobalProvider>
  );
  return { ...renderAPI, mockReducer: combinedReducers, deps };
};
