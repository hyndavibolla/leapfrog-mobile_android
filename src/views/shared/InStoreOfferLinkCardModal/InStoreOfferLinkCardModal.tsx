import React, { memo } from 'react';
import { View, Text } from 'react-native';

import { styles } from './styles';
import { useTestingHelper } from '../../../utils/useTestingHelper';
import { Title, TitleType } from '../Title';
import { Button } from '../Button';
import CardIcon from '../../../assets/shared/cardIconDarkBlue.svg';
import RestaurantFallback from '../../../assets/in-store-offer/restaurantFallback.svg';
import { BrandLogo } from '../BrandLogo';

export interface Props {
  brandLogo?: string;
  brandName: string;
  onLinkCardPress?: () => void;
  onCancel?: () => void;
}

export const InStoreOfferLinkCardModalKey = 'in-store-offer-link-card-modal';

const InStoreOfferLinkCardModal = ({ brandLogo, brandName, onLinkCardPress, onCancel }: Props) => {
  const { getTestIdProps } = useTestingHelper(InStoreOfferLinkCardModalKey);
  return (
    <View style={styles.container} {...getTestIdProps('container')}>
      <View style={styles.header}>
        <CardIcon style={styles.brandIcon} />
        <Text style={styles.plus}>+</Text>
        <BrandLogo image={brandLogo} Fallback={RestaurantFallback} size={60} />
      </View>
      <Title style={styles.title} type={TitleType.HEADER}>
        To activate your {brandName} offer, link your card
      </Title>
      <View style={styles.body}>
        <Text style={styles.bodyText}>Please add a card to your account to start earning points at restaurants near you.</Text>
      </View>
      <Button innerContainerStyle={styles.linkBtnInnerContainer} textStyle={styles.linkBtnText} onPress={onLinkCardPress} {...getTestIdProps('link-card-btn')}>
        Link a Card
      </Button>
      <Button innerContainerStyle={styles.dismissBtnInnerContainer} textStyle={styles.dismissBtnText} onPress={onCancel} {...getTestIdProps('cancel-btn')}>
        Cancel
      </Button>
    </View>
  );
};

export default memo(InStoreOfferLinkCardModal);
