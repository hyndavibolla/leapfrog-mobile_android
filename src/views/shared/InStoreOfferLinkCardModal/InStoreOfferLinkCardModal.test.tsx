import React from 'react';
import { fireEvent } from '@testing-library/react-native';
import { GeneralModel } from '../../../models';
import { getMission_1 } from '../../../test-utils/entities';
import { getMockDeps } from '../../../test-utils/getMockDeps';
import { renderWithGlobalContext } from '../../../test-utils/renderWithGlobalContext';

import InStoreOfferLinkCardModal, { Props, InStoreOfferLinkCardModalKey } from './InStoreOfferLinkCardModal';

describe('InStoreOfferLinkCardModal', () => {
  let deps: GeneralModel.Deps;
  let props: Props;

  beforeEach(() => {
    deps = getMockDeps();
    const { brandName, brandLogo } = getMission_1();
    props = {
      brandName,
      brandLogo,
      onLinkCardPress: jest.fn(),
      onCancel: jest.fn()
    };
  });

  it('should render', () => {
    const { toJSON } = renderWithGlobalContext(<InStoreOfferLinkCardModal {...props} />, deps);
    expect(toJSON()).toMatchSnapshot();
  });

  it('should render without brand logo', () => {
    props.brandLogo = null;
    const { toJSON } = renderWithGlobalContext(<InStoreOfferLinkCardModal {...props} />, deps);
    expect(toJSON()).toMatchSnapshot();
  });

  it('should have a pressable link card button', () => {
    const { getByTestId } = renderWithGlobalContext(<InStoreOfferLinkCardModal {...props} />, deps);
    fireEvent.press(getByTestId(`${InStoreOfferLinkCardModalKey}-link-card-btn`));
    expect(props.onLinkCardPress).toBeCalled();
  });

  it('should have a pressable cancel button', () => {
    const { getByTestId } = renderWithGlobalContext(<InStoreOfferLinkCardModal {...props} />, deps);
    fireEvent.press(getByTestId(`${InStoreOfferLinkCardModalKey}-cancel-btn`));
    expect(props.onCancel).toBeCalled();
  });
});
