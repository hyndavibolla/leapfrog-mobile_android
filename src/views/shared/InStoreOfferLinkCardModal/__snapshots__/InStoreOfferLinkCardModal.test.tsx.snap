// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InStoreOfferLinkCardModal should render 1`] = `
<View
  accessibilityLabel="in-store-offer-link-card-modal-container"
  accessible={true}
  style={
    Object {
      "alignItems": "center",
    }
  }
  testID="in-store-offer-link-card-modal-container"
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "center",
        "marginBottom": 24,
      }
    }
  >
    CardIconDarkBlue
    <Text
      style={
        Object {
          "color": "#262C2F",
          "fontFamily": "SFProDisplay-Heavy",
          "fontSize": 22,
          "lineHeight": 26,
          "marginBottom": 0,
          "marginLeft": 12,
          "marginRight": 12,
          "marginTop": 0,
        }
      }
    >
      +
    </Text>
    <View
      style={
        Array [
          Object {
            "height": 60,
            "width": 60,
          },
          undefined,
        ]
      }
    >
      <View
        style={
          Array [
            Object {
              "overflow": "hidden",
            },
            Object {
              "alignItems": "center",
              "borderBottomLeftRadius": 60,
              "borderBottomRightRadius": 60,
              "borderTopLeftRadius": 60,
              "borderTopRightRadius": 60,
              "height": "100%",
              "justifyContent": "center",
              "width": "100%",
            },
          ]
        }
      >
        <FastImageView
          accessibilityLabel="brand-logo-image"
          accessible={true}
          onFastImageError={[Function]}
          onFastImageLoad={[Function]}
          onFastImageLoadEnd={[Function]}
          onFastImageLoadStart={[Function]}
          resizeMode="cover"
          source={
            Object {
              "priority": "high",
              "uri": "https://www.movistar.es/estaticos/img/ico-app-mi-movistar-2017.png",
            }
          }
          style={
            Object {
              "bottom": 0,
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
            }
          }
          testID="brand-logo-image"
        />
      </View>
    </View>
  </View>
  <Text
    style={
      Array [
        Object {
          "color": "#262C2F",
          "fontFamily": "SFProDisplay-Bold",
          "fontSize": 20,
          "lineHeight": 30,
        },
        Object {
          "color": "#262C2F",
          "textAlign": "center",
        },
      ]
    }
    type="titleHeader"
  >
    To activate your 
    Starbucks
     offer, link your card
  </Text>
  <View
    style={
      Object {
        "marginTop": 8,
      }
    }
  >
    <Text
      style={
        Object {
          "fontSize": 16,
          "textAlign": "center",
        }
      }
    >
      Please add a card to your account to start earning points at restaurants near you.
    </Text>
  </View>
  <View
    accessibilityLabel="in-store-offer-link-card-modal-link-card-btn"
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
    testID="in-store-offer-link-card-modal-link-card-btn"
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "backgroundColor": "#0066CC",
            "borderBottomLeftRadius": 30,
            "borderBottomRightRadius": 30,
            "borderTopLeftRadius": 30,
            "borderTopRightRadius": 30,
            "flexDirection": "row",
            "justifyContent": "center",
            "paddingBottom": 8,
            "paddingLeft": 12,
            "paddingRight": 12,
            "paddingTop": 8,
          },
          Object {
            "marginTop": 40,
            "paddingBottom": 16,
            "paddingLeft": 100,
            "paddingRight": 100,
            "paddingTop": 16,
          },
        ]
      }
    >
      <View
        style={false}
      >
        <Text
          style={
            Array [
              Object {
                "color": "#909BA0",
                "fontFamily": "SFProDisplay-Medium",
                "fontSize": 16,
                "lineHeight": 22,
              },
              Array [
                Object {
                  "color": "#FFFFFF",
                  "fontSize": 14,
                  "textAlign": "center",
                },
                Object {
                  "fontFamily": "SFProDisplay-Medium",
                  "fontSize": 16,
                },
                undefined,
              ],
            ]
          }
        >
          Link a Card
        </Text>
      </View>
    </View>
  </View>
  <View
    accessibilityLabel="in-store-offer-link-card-modal-cancel-btn"
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
    testID="in-store-offer-link-card-modal-cancel-btn"
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "backgroundColor": "#0066CC",
            "borderBottomLeftRadius": 30,
            "borderBottomRightRadius": 30,
            "borderTopLeftRadius": 30,
            "borderTopRightRadius": 30,
            "flexDirection": "row",
            "justifyContent": "center",
            "paddingBottom": 8,
            "paddingLeft": 12,
            "paddingRight": 12,
            "paddingTop": 8,
          },
          Object {
            "backgroundColor": "transparent",
            "marginTop": 16,
          },
        ]
      }
    >
      <View
        style={false}
      >
        <Text
          style={
            Array [
              Object {
                "color": "#909BA0",
                "fontFamily": "SFProDisplay-Medium",
                "fontSize": 16,
                "lineHeight": 22,
              },
              Array [
                Object {
                  "color": "#FFFFFF",
                  "fontSize": 14,
                  "textAlign": "center",
                },
                Object {
                  "color": "#0066CC",
                  "fontFamily": "SFProDisplay-Bold",
                  "fontSize": 16,
                },
                undefined,
              ],
            ]
          }
        >
          Cancel
        </Text>
      </View>
    </View>
  </View>
</View>
`;

exports[`InStoreOfferLinkCardModal should render without brand logo 1`] = `
<View
  accessibilityLabel="in-store-offer-link-card-modal-container"
  accessible={true}
  style={
    Object {
      "alignItems": "center",
    }
  }
  testID="in-store-offer-link-card-modal-container"
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "center",
        "marginBottom": 24,
      }
    }
  >
    CardIconDarkBlue
    <Text
      style={
        Object {
          "color": "#262C2F",
          "fontFamily": "SFProDisplay-Heavy",
          "fontSize": 22,
          "lineHeight": 26,
          "marginBottom": 0,
          "marginLeft": 12,
          "marginRight": 12,
          "marginTop": 0,
        }
      }
    >
      +
    </Text>
    <View
      style={
        Array [
          Object {
            "height": 60,
            "width": 60,
          },
          undefined,
        ]
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "height": "100%",
            "justifyContent": "center",
            "width": "100%",
          }
        }
      >
        RestaurantFallback
      </View>
    </View>
  </View>
  <Text
    style={
      Array [
        Object {
          "color": "#262C2F",
          "fontFamily": "SFProDisplay-Bold",
          "fontSize": 20,
          "lineHeight": 30,
        },
        Object {
          "color": "#262C2F",
          "textAlign": "center",
        },
      ]
    }
    type="titleHeader"
  >
    To activate your 
    Starbucks
     offer, link your card
  </Text>
  <View
    style={
      Object {
        "marginTop": 8,
      }
    }
  >
    <Text
      style={
        Object {
          "fontSize": 16,
          "textAlign": "center",
        }
      }
    >
      Please add a card to your account to start earning points at restaurants near you.
    </Text>
  </View>
  <View
    accessibilityLabel="in-store-offer-link-card-modal-link-card-btn"
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
    testID="in-store-offer-link-card-modal-link-card-btn"
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "backgroundColor": "#0066CC",
            "borderBottomLeftRadius": 30,
            "borderBottomRightRadius": 30,
            "borderTopLeftRadius": 30,
            "borderTopRightRadius": 30,
            "flexDirection": "row",
            "justifyContent": "center",
            "paddingBottom": 8,
            "paddingLeft": 12,
            "paddingRight": 12,
            "paddingTop": 8,
          },
          Object {
            "marginTop": 40,
            "paddingBottom": 16,
            "paddingLeft": 100,
            "paddingRight": 100,
            "paddingTop": 16,
          },
        ]
      }
    >
      <View
        style={false}
      >
        <Text
          style={
            Array [
              Object {
                "color": "#909BA0",
                "fontFamily": "SFProDisplay-Medium",
                "fontSize": 16,
                "lineHeight": 22,
              },
              Array [
                Object {
                  "color": "#FFFFFF",
                  "fontSize": 14,
                  "textAlign": "center",
                },
                Object {
                  "fontFamily": "SFProDisplay-Medium",
                  "fontSize": 16,
                },
                undefined,
              ],
            ]
          }
        >
          Link a Card
        </Text>
      </View>
    </View>
  </View>
  <View
    accessibilityLabel="in-store-offer-link-card-modal-cancel-btn"
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
    testID="in-store-offer-link-card-modal-cancel-btn"
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "backgroundColor": "#0066CC",
            "borderBottomLeftRadius": 30,
            "borderBottomRightRadius": 30,
            "borderTopLeftRadius": 30,
            "borderTopRightRadius": 30,
            "flexDirection": "row",
            "justifyContent": "center",
            "paddingBottom": 8,
            "paddingLeft": 12,
            "paddingRight": 12,
            "paddingTop": 8,
          },
          Object {
            "backgroundColor": "transparent",
            "marginTop": 16,
          },
        ]
      }
    >
      <View
        style={false}
      >
        <Text
          style={
            Array [
              Object {
                "color": "#909BA0",
                "fontFamily": "SFProDisplay-Medium",
                "fontSize": 16,
                "lineHeight": 22,
              },
              Array [
                Object {
                  "color": "#FFFFFF",
                  "fontSize": 14,
                  "textAlign": "center",
                },
                Object {
                  "color": "#0066CC",
                  "fontFamily": "SFProDisplay-Bold",
                  "fontSize": 16,
                },
                undefined,
              ],
            ]
          }
        >
          Cancel
        </Text>
      </View>
    </View>
  </View>
</View>
`;
